src/dotty/tools/dotc/core/TypeComparer.scala

```diff
@@ -524,13 +560,144 @@ class TypeComparer(initctx: Context) extends DotClass with ConstraintHandling {
       false
   }
 
+  /** Subtype test for the hk application `tp2 = tycon2[args2]`.
+   */
+  def compareHkApply2(tp1: Type, tp2: Type, tycon2: Type, args2: List[Type]): Boolean = {
+    val tparams = tycon2.typeParams
+    assert(tparams.nonEmpty)
+
+    def isMatchingApply(tp1: Type): Boolean = tp1 match {
+      case HKApply(tycon1, args1) =>
```

Comment:
So if I have something like compareHkApply2(`List { type List$$A = Int }`, `CC[Int]`, `CC`, `Int`), where `CC` is `[X] -> List[X]`, it's expected to fail and be handled somewhere else? Could this be documented?